### PR TITLE
Add support for NewPing

### PR DESCRIPTION
**Pull request by ruimarinho**
* Works with many different ultrasonic sensor models: HC-SR04, SRF05,  SRF06, DYP-ME007, JSN-SR04T & Parallax PING)))™.
* Built-in digital filter method ping_median() for easy error correction (use `SONAR_ITERATIONS` build flag to configure).
* Allows setting of a maximum distance where pings beyond that distance are read as no ping or clear (use `SONAR_MAX_DISTANCE` build flag).

### CODE REVIEW

**Comment by mcspr** [comment]
Quick note: TRAVIS02 (in hardware.h) still has HCSR04_SUPPORT.

File: code/espurna/sensor.ino [hunk 1]
```diff
@@ -476,11 +476,13 @@ void _sensorLoad() {
...
 
-    #if HCSR04_SUPPORT
+    #if SONAR_SUPPORT
     {
-        HCSR04Sensor * sensor = new HCSR04Sensor();
-        sensor->setTrigger(HCSR04_TRIGGER);
-        sensor->setEcho(HCSR04_ECHO);
+        SonarSensor * sensor = new SonarSensor();
+        sensor->setEcho(SONAR_ECHO);
+        sensor->setIterations(SONAR_PING_ITERATIONS);
```
Comment (mcspr) [inline code/espurna/sensor.ino]: Typo. Should be `SONAR_ITERATIONS`

File: code/espurna/config/hardware.h [hunk 2]
```diff
@@ -112,6 +112,9 @@
     #define I2C_SDA_PIN         12  // D6
     #define I2C_SCL_PIN         14  // D5
 
+    #define SONAR_TRIGGER       12  // D6
+    #define SONAR_ECHO          13  // D7
+
```
Comment (mcspr) [inline code/espurna/config/hardware.h]: Maybe just keep as defaults in sensor.h?

Comment (xoseperez) [reply code/espurna/config/hardware.h]: Agree, these should not be here.

File: code/espurna/sensor.ino (hunk 1 above)
Comment (xoseperez) [reply code/espurna/sensor.ino]: Right

File: code/espurna/config/hardware.h (hunk 2 above)
Comment (mcspr) [reply code/espurna/config/hardware.h]: Oops. Pins from wemos do not include hw.h 7 and 8 (6-11 are not mapped to pins at all so they do nothing): https://github.com/esp8266/Arduino/blob/master/variants/d1/pins_arduino.h
What i meant to say, change the default to these and just not define it only for wemos hw :) Sry for confusion

Comment (ruimarinho) [reply code/espurna/config/hardware.h]: @xoseperez I don't think these will work on Wemos — https://github.com/xoseperez/espurna/blob/dev/code/espurna/config/hardware.h#L2669-L2671. Wasn't this change correct after all?

Comment (mcspr) [reply code/espurna/config/hardware.h]: @ruimarinho nah you linked travis-only hw that is only a built test, no real boards involved. these are fine https://github.com/xoseperez/espurna/blob/4f61d3aa7e2e6526fdb518a9f84785b055ac0ab7/code/espurna/config/sensors.h#L570-L576

**Pull request merged**